internal/controllers/core/tiltfile/buildsource.go

```diff
@@ -0,0 +1,69 @@
+package tiltfile
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/tilt-dev/tilt/internal/store"
+	"github.com/tilt-dev/tilt/pkg/model"
+	"github.com/tilt-dev/tilt/pkg/model/logstore"
+)
+
+// BuildSource is a way for legacy Tilt reconcilers to
+// notify the reconciliation loop that a Tiltfile needs to be rebuilt.
+//
+// We model this as a controller-runtime Source, which is intended for watching
+// external resources that trigger reconciliation.
+
+type BuildEntry struct {
+	Name                  model.ManifestName
+	FilesChanged          []string
+	BuildReason           model.BuildReason
+	UserConfigState       model.UserConfigState
+	TiltfilePath          string
+	CheckpointAtExecStart logstore.Checkpoint
+	EngineMode            store.EngineMode
+}
+
+type BuildSource struct {
+	mu    sync.Mutex
+	entry *BuildEntry
+	q     workqueue.RateLimitingInterface
+}
+
+var _ source.Source = &BuildSource{}
+
+func NewBuildSource() *BuildSource {
+	return &BuildSource{}
+}
+
+func (s *BuildSource) Start(ctx context.Context, handler handler.EventHandler, q workqueue.RateLimitingInterface, ps ...predicate.Predicate) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.q = q
+	return nil
+}
+
+func (s *BuildSource) Entry() *BuildEntry {
```

Comment:
AFAICT this is unused - what's the reason for holding onto the latest vs just having an `AddEntry()` and writing to the queue?

Reply:
yep - it will be used in the next PR : )